purchase_last_price_info/models/product.py

```diff
@@ -0,0 +1,39 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, fields
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    def _get_last_purchase(self):
+        """
+        Get last purchase price, last purchase date and last customer
+        """
+        for product in self:
+            last_date = False
+            last_price = False
+            last_seller = False
+            lines = self.env['purchase.order.line'].search(
+                [('product_id', '=', product.id),
+                 ('state', 'in', ['confirmed', 'done'])])
+            if lines:
+                old_lines = sorted(lines, key=lambda l: l.order_id.date_order,
+                                   reverse=True)
+                last_date = old_lines[0].order_id.date_order
```

Comment:
If you make `old_line[:1]`, you remove the need of the condition, the initialization to False, and so on.